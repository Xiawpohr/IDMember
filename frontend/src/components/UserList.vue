<template>
  <v-container fluid grid-list-lg>
    <v-data-iterator
        :items="users"
        :rows-per-page-items="rowsPerPageItems"
        :pagination.sync="pagination"
        content-tag="v-layout"
        row
        wrap
    >
      <v-flex
        slot="item"
        slot-scope="props"
        xs12
        sm6
        md4
        lg3
      >
        <UserItem :user="props.item" />
      </v-flex>
    </v-data-iterator>
  </v-container>
</template>

<script>
import UserItem from './UserItem.vue'

export default {
  components: {
    UserItem
  },
  props: {
    users: {
      type: Array,
      required: true,
      default: () => []
    }
  },
  data() {
    return {
      rowsPerPageItems: [8, 12, 16],
      pagination: {
        rowsPerPage: 8
      },
    }
  }
}
</script>
