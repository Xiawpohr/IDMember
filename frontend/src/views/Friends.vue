<template>
  <div>
    <h1>Your friends are here! You can give them some money</h1>
    <UserList :users="friends"/>
  </div>
</template>

<script>
import UserList from '@/components/UserList.vue'

export default {
  components: {
    UserList
  },
  computed: {
    friends() {
      return this.$store.state.friend.friends
    }
  },
  methods: {
    fetchFriends() {
      this.$store.dispatch('friend/fetchAll')
    }
  },
  created () {
    this.fetchFriends()
  }
}
</script>
