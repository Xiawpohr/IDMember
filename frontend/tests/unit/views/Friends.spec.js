import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import Friends from '@/views/Friends.vue'
import UserList from '@/components/UserList.vue'

const localVue = createLocalVue()
localVue.use(Vuex)

let wrapper, state, actions, store

beforeEach(() => {
  state = {
    friends: [{ name: 'Arthur' }, { name: 'Robert' }]
  }
  actions = {
    fetchAll: jest.fn()
  }
  store = new Vuex.Store({
    modules: {
      friend: {
        namespaced: true,
        state,
        actions
      }
    }
  })
  wrapper = shallowMount(Friends, {
    localVue,
    store
  })
})

it('should render the component', () => {
  expect(wrapper.html()).toMatchSnapshot()
})

it('should render UserList component', () => {
  const userList = wrapper.find(UserList)
  expect(userList.exists()).toBe(true)
})

it('should pass users prop to UserList component', () => {
  const userList = wrapper.find(UserList)
  expect(userList.vm.users).toBe(state.friends)
})

it('should fetch all friends when the component created', () => {
  expect(actions.fetchAll).toHaveBeenCalled()
})
