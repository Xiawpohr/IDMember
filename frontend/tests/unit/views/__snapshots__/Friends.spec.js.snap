// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the component 1`] = `
<div>
  <h1>Your friends are here! You can give them some money</h1>
  <userlist-stub users="[object Object],[object Object]"></userlist-stub>
</div>
`;
